include/mfmg/cuda/cuda_matrix_operator.cuh

```diff
@@ -50,7 +51,8 @@ public:
   std::shared_ptr<SparseMatrixDevice<value_type>> get_matrix() const;
 
 private:
-  std::shared_ptr<SparseMatrixDevice<value_type>> _matrix;
+  mutable std::shared_ptr<SparseMatrixDevice<value_type>> _matrix;
+  mutable std::shared_ptr<SparseMatrixDevice<value_type>> _transposed_matrix;
```

Comment:
Why do you need to store both matrices?

Reply:
The goal was to try to minimize the number of transpose calculations. Transpose actually appears in multiple places (through `apply(TRANS)` and `transpose()`, for example), so by storing it we essentially cache the calculation.